### ISSUE TITLE
Keep downloaded agent binary until agent upgrade is completed

### ISSUE THREAD

**Issue by juliaElastic** [elastic/elastic-agent#6191]
Related to https://github.com/elastic/ingest-dev/issues/2878

During agent upgrade, there could be failures that cause the upgrade to be retried. When the retries happen, the agent binary should not be downloaded multiple times.
Keep the downloaded agent binary until the upgrade is completed successfully, and delete when the upgrade is completed.

**Comment by cmacknz** [comment]
I think this may fall out of the rollback implementation which should introduce a TTL for old agent versions. I don't think we should work on this until that RFC is completed so we can see where it fits in. Moving to blocked.

- https://github.com/elastic/kibana/issues/172745

**Comment by nimarezainia** [comment]
moving this out of sprint40 for now.

**Comment by nimarezainia** [comment]
@pchila @cmacknz  do we need this issue open? will it be addressed already with the current rollback implementation?

**Comment by cmacknz** [comment]
Yes, this is suggesting that we keep the new agent version on disk util we complete the upgrade. The rollback is keeping the previous version of the agent until a TTL expires.

**Comment by ebeahan** [comment]
> I don't think we should work on this until that RFC is completed so we can see where it fits in. Moving to blocked.

Do we still consider this blocked by any other work?

**Comment by cmacknz** [comment]
We should wait until the upgrade rollback implementation is complete with its current feature set before looking at this one.

**Comment by nimarezainia** [comment]
rollback in 9.3 addresses this concern.
